### ISSUE TITLE
No balance between projects in election

### ISSUE THREAD

**Issue by g4s8** [zerocracy/farm#1174]
If user is contributing to many projects with Zerocracy with people deficit (many jobs in WBS which are not assigned to anyone), most probably he/she will be assigned only to jobs in a project with "early ping", because usually there will be only one slot for job in the agenda. We need to fix election process somehow, so user should be assigned to jobs from different project in a balanced way.

Let's make new `Votes` implementation which will count all user's project and vote for user who has less jobs in electing project than others:
```
vote = ((jobs in current project) / (total jobs)) * (total projects)
```

**Comment by 0crat** [comment]
Job [#1174](https://github.com/zerocracy/farm/issues/1174) is now in scope, role is `DEV`

**Comment by 0crat** [comment]
Bug was reported, see [§29](http://www.zerocracy.com/policy.html#29): +15 point(s) just awarded to @g4s8[/z](https://www.0crat.com/u/g4s8)

**Comment by g4s8** [comment]
@0crat assign @krzyk

**Comment by 0crat** [comment]
> @0crat assign @krzyk  ([here](https://github.com/zerocracy/farm/issues/1174#issuecomment-399749460))

@g4s8 The job [#1174](https://github.com/zerocracy/farm/issues/1174) assigned to @krzyk[/z](https://www.0crat.com/u/krzyk), here is [why](https://www.0crat.com/footprint/C3NDPUA8L/1751136371); the budget is 30 minutes, see [§4](http://www.zerocracy.com/policy.html#4); please, read [§8](http://www.zerocracy.com/policy.html#8) and [§9](http://www.zerocracy.com/policy.html#9); if the task is not clear, read [this](http://www.yegor256.com/2015/02/16/it-is-not-a-school.html) and [this](http://www.yegor256.com/2015/01/15/how-to-cut-corners.html); there will be a monetary reward for this job

**Comment by 0crat** [comment]
Manual assignment of issues is discouraged, see [§19](http://www.zerocracy.com/policy.html#19): -5 point(s) just awarded to @g4s8[/z](https://www.0crat.com/u/g4s8)

**Comment by g4s8** [comment]
@krzyk the idea here is to downvote people who has higher task ratio in particular project, by returning value from 0.0 to `n`, where `n` means high ratio (all jobs from one project in `n` projects), 1 - balanced ratio (e.g. 2 of 4 jobs from 2 projects or all jobs from one project) 0 - low ratio (no jobs from the project, but has jobs from other projects), then we can multiply it for some negative coefficient in `elect_performer`, to raise people with low ration upper.

**Comment by krzyk** [comment]
@g4s8 OK, but all voters are usually decorated with VsSafe, which checks if the value returned is in 0.0 - 1.0 range.

And the suggested equation will generate values 0.0 - number_of_user_projects, shouldn't it be normalized?

**Comment by g4s8** [comment]
@krzyk yes, looks like we have to normalize it. So we need to know max project count among all possible job performers before starting election.

**Comment by krzyk** [comment]
@g4s8 please close this issue, it has been merged in #1256

**Comment by g4s8** [comment]
@krzyk thanks

**Comment by 0crat** [comment]
@ypshenychka[/z](https://www.0crat.com/u/ypshenychka) please review this job completed by @krzyk[/z](https://www.0crat.com/u/krzyk), as in [§30](http://www.zerocracy.com/policy.html#30); the job will be fully closed and all payments will be made when the quality review is completed

**Comment by 0crat** [comment]
The job [#1174](https://github.com/zerocracy/farm/issues/1174) is now out of scope

**Comment by ypshenychka** [comment]
@0crat quality good

**Comment by 0crat** [comment]
Order was finished, quality is "good": +38 point(s) just awarded to @krzyk[/z](https://www.0crat.com/u/krzyk)

**Comment by 0crat** [comment]
Quality review completed: +8 point(s) just awarded to @ypshenychka[/z](https://www.0crat.com/u/ypshenychka)
